ext/dom/html5_parser.c

```diff
@@ -0,0 +1,262 @@
+            size_t public_id_len, system_id_len;
+            const lxb_char_t *public_id = lxb_dom_document_type_public_id(doctype, &public_id_len);
+            const lxb_char_t *system_id = lxb_dom_document_type_system_id(doctype, &system_id_len);
+            xmlDtdPtr lxml_dtd = xmlCreateIntSubset(lxml_doc, name, public_id_len ? public_id : NULL, system_id_len ? system_id : NULL);
+            if (UNEXPECTED(lxml_dtd == NULL)) {
+                retval = LEXBOR_LIBXML2_BRIDGE_STATUS_OOM;
+                goto out;
+            }
+            /* libxml2 doesn't support line numbers on this anyway, it returns -1 instead, so don't bother */
+        } else if (node->type == LXB_DOM_NODE_TYPE_ATTRIBUTE) {
+            lxb_dom_attr_t *attr = lxb_dom_interface_attr(node);
+            do {
+                /* Same namespace remark as for elements */
+                const lxb_char_t *local_name = lxb_dom_attr_local_name(attr, NULL);
+                const lxb_char_t *value = lxb_dom_attr_value(attr, NULL);
+                xmlAttrPtr lxml_attr = xmlSetNsProp(lxml_parent, NULL, local_name, value);
+                if (UNEXPECTED(lxml_attr == NULL)) {
+                    retval = LEXBOR_LIBXML2_BRIDGE_STATUS_OOM;
+                    goto out;
+                }
+                attr = attr->next;
+                /* libxml2 doesn't support line numbers on this anyway, it derives them instead, so don't bother */
+            } while (attr);
+        } else if (node->type == LXB_DOM_NODE_TYPE_COMMENT) {
+            lxb_dom_comment_t *comment = lxb_dom_interface_comment(node);
+            xmlNodePtr lxml_comment = xmlNewDocComment(lxml_doc, comment->char_data.data.data);
+            if (UNEXPECTED(lxml_comment == NULL)) {
+                retval = LEXBOR_LIBXML2_BRIDGE_STATUS_OOM;
+                goto out;
+            }
+            xmlAddChild(lxml_parent, lxml_comment);
+            lxml_comment->line = sanitize_line_nr(node->line);
+        }
+    }
+
+out:
+    lexbor_array_obj_destroy(&work_list, false);
+    return retval;
+}
+
+void lexbor_libxml2_bridge_parse_context_init(lexbor_libxml2_bridge_parse_context *ctx)
+{
+    memset(ctx, 0, sizeof(*ctx));
+}
+
+void lexbor_libxml2_bridge_parse_set_error_callbacks(lexbor_libxml2_bridge_parse_context *ctx, lexbor_libxml2_bridge_tokenizer_error_reporter tokenizer_error_reporter, lexbor_libxml2_bridge_tree_error_reporter tree_error_reporter)
+{
+    ctx->tokenizer_error_reporter = tokenizer_error_reporter;
+    ctx->tree_error_reporter = tree_error_reporter;
+}
+
+lexbor_libxml2_bridge_status lexbor_libxml2_bridge_convert_document(lxb_html_document_t *document, xmlDocPtr *doc_out, bool compact_text_nodes, bool create_default_ns)
+{
+#ifdef LIBXML_HTML_ENABLED
+    xmlDocPtr lxml_doc = htmlNewDocNoDtD(NULL, NULL);
+#else
+    xmlDocPtr lxml_doc = xmlNewDoc((const xmlChar *) "1.0");
+    lxml_doc->type = XML_HTML_DOCUMENT_NODE;
+#endif
```

Comment:
I'm not totally sure I get the ifdef here?

Is this to switch to libxml2 HTML parsing if it is enabled?